### PR TITLE
Throw StrictDynamicMappingException exception if `dynamic:strict` and undeclared field value is `null`

### LINKED ISSUE #9444: Dynamic "strict" not working if an undeclared field has null value
Hello.
I have created type mapping with dynamic 'strict'. And I found that a StrictDynamicMappingException exception is not thrown when I try to index a document with an undeclared field and it's value NULL 

##### delete the index
<code>curl -XDELETE localhost:9200/test</code>

##### create the index with mapping and dynamic "strict"
<code>curl -XPOST localhost:9200/test -d '{
    "mappings" : {
        "type1" : {
            "dynamic": "strict",
            "properties" : {
                "field1" : { "type" : "string", "index" : "not_analyzed" }
            }
        }
    }
}'</code>

##### index a document with declared fields - ok
<code>curl -XPUT localhost:9200/test/type1/1 -d '
{ "field1": "test document"}
'</code>

##### index a document with undeclared field - ok,  we have error <code>StrictDynamicMappingException</code>

<code>curl -XPUT localhost:9200/test/type1/1 -d '
{ "field1": "test document",
"field2": "test document"},
'</code>

##### index a document with undeclared field (field2) and it's value NULL,  we have no error <code>StrictDynamicMappingException</code>
<code>curl -XPUT localhost:9200/test/type1/2 -d '
{ "field1": "test document",
"field2": null},
'</code>

##### there is a new (undeclared) field (field2) 
<code>curl -XGET localhost:9200/test/type1/_search -d '
{
    "query": {
        "match_all": {}
    }
}'</code>

### PR DESCRIPTION
**Pull request by simaov**
Throw StrictDynamicMappingException exception if dynamic is 'strict' and undeclared field value is NULL, test for this

Closes #9444

### CODE REVIEW

File: src/main/java/org/elasticsearch/index/mapper/object/ObjectMapper.java [hunk 1]
```diff
@@ -587,6 +587,10 @@ private void serializeNullValue(ParseContext context, String lastFieldName) thro
                 }
             }
             mapper.parse(context);
+        } else {
+            if (dynamic == Dynamic.STRICT) {
```
Comment (rjernst) [inline src/main/java/org/elasticsearch/index/mapper/object/ObjectMapper.java]: This can be joined together with the `else` into an `else if`.

**Comment by rjernst** [comment]
@simaov Thanks for the fix! I had just one minor comment.

While you are making that change, could I please ask you to sign our CLA so we can merge this in?
http://www.elasticsearch.org/contributor-agreement/

**Comment by clintongormley** [comment]
@rjernst CLA has been signed - could you merge?

**Comment by rjernst** [comment]
Thanks @simaov!

**Pull request closed without merging**